rpcs3/rpcs3qt/emu_settings.cpp

```diff
@@ -307,6 +307,49 @@ void emu_settings::EnhanceCheckBox(QCheckBox* checkbox, emu_settings_type type)
 	});
 }
 
+void emu_settings::EnhanceDateTimeEdit(QDateTimeEdit* date_time_edit, emu_settings_type type, const QString& format, bool calendarMode)
+{
+	if (!date_time_edit)
+	{
+		cfg_log.fatal("EnhanceDateTimeEdit '%s' was used with an invalid object", cfg_adapter::get_setting_name(type));
+		return;
+	}
+
+	date_time_edit->setDisplayFormat(format);
+	date_time_edit->setCalendarPopup(calendarMode);
+
+	bool ok_def = false;
+	const s64 def = qstr(GetSettingDefault(type)).toLongLong(&ok_def);
+	if (!ok_def)
+	{
+		cfg_log.fatal("EnhanceDateTimeEdit '%s' was used with an invalid emu_settings_type", cfg_adapter::get_setting_name(type));
+		return;
+	}
+
+	bool ok_sel = false;
+	s64 val = qstr(GetSetting(type)).toLongLong(&ok_sel);
+
+	if (!ok_sel)
+	{
+		cfg_log.error("EnhanceDateTimeEdit '%s' tried to set an invalid value: %d. Setting to default: %d", cfg_adapter::get_setting_name(type), val, def);
+		val = def;
+		m_broken_types.insert(type);
+		SetSetting(type, sstr(def));
+	}
+
+	// PS3 RTC ticks are in microseconds, so we'll store our times in this format, but QT likes milliseconds, so we throw out the little remainder
+	QDateTime game_time = QDateTime::currentDateTime().addSecs(val);
+	date_time_edit->setDateTime(game_time);
+
+	connect(date_time_edit, &QDateTimeEdit::dateTimeChanged,  [date_time_edit, type, this](QDateTime value)
+	{
+		// PS3 RTC is in microseconds, so we scale up the time difference to match
+		s64 offset = QDateTime::currentDateTime().secsTo(value);
+		SetSetting(type, sstr(offset));
```

Comment:
why sstr. We only use sstr for conversions between watering and std string

Reply:
'watering'?
The sstr is just from it's use in the immediately below EnhanceSlider function, where it's used to convert an int -> std::string.
I assumed it was a common path to 'settings string' (sstr?)
Should I be formatting into a string directly instead? (isn't that the job of the serialisation process??)

Reply:
It's just a random conversion. Doesn't std::to_string work with s64 ?

Reply:
I meant qstring. Autocorrect.

Reply:
I don't know why it was used elsewhere. Probably out of laziness

Reply:
The conversion may have worked if you used qint64. Which is the proper type of secTo.

Reply:
You were right on the std::to_string, it's a std::string that SetSetting wants.
Yeah, if I static_cast to qlonglong it was happy.  But std::to_string will fit the bill.
I still feel that we're doing this setting stuff wrong. I think we're throwing out type info too early (but it seems pretty locked in now).  I actually thought YAML would have had a base level serialisation for standard variables, and then a class to inherit and override for custom classes / structures.  That's how I'm used to it in the C# world at least.